src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -142,8 +146,8 @@ trait Erasure {
           // this replaces each typeref that refers to an argument
           // by the type `p.tpe` of the actual argument p (p in params)
           else apply(mt.resultType(mt.paramTypes)))
-      case RefinedType(parents, decls) =>
-        apply(mergeParents(parents))
+      case rt : RefinedType =>
+        apply(mergeParents(rt.normalize.parents)) // RefinedType#normalize
```

Comment:
Would it make sense to move RefinedType's `normalizeImpl` here? Also, could we do less work when doing Java-style erasure?